src/Mailgun/Api/Domains.php

```diff
@@ -0,0 +1,270 @@
+    }
+
+    /**
+     * Creates a new domain for the account.
+     * See below for spam filtering parameter information.
+     * {@link https://documentation.mailgun.com/user_manual.html#um-spam-filter}
+     *
+     * @param string    $domain     Name of the domain.
+     * @param string    $smtpPass   Password for SMTP authentication.
+     * @param string    $spamAction `disable` or `tag` - inbound spam filtering.
+     * @param bool      $wildcard   Domain will accept email for subdomains.
+     *
+     * @return ComplexDomain
+     */
+    public function create($domain, $smtpPass, $spamAction, $wildcard)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::stringNotEmpty($smtpPass);
+        // TODO(sean.johnson): Extended spam filter input validation.
+        Assert::stringNotEmpty($spamAction);
+        Assert::boolean($wildcard);
+
+        $params = array(
+            'name'          => $domain,
+            'smtp_password' => $smtpPass,
+            'spam_action'   => $spamAction,
+            'wildcard'      => $wildcard,
+        );
+
+        $response = $this->post('/v3/domains', $params);
+
+        return $this->serializer->deserialize($response, ComplexDomain::class);
+    }
+
+    /**
+     * Removes a domain from the account.
+     * WARNING: This action is irreversible! Be cautious!
+     *
+     * @param string    $domain     Name of the domain.
+     *
+     * @return ...
+     */
+    public function remove($domain)
+    {
+        Assert::stringNotEmpty($domain);
+
+        $response = $this->delete(sprintf('/v3/domains/%s', $domain));
+
+        return $this->serializer->deserialize($response, SimpleResponse::class);
+    }
+
+    /**
+     * Returns a list of SMTP credentials for the specified domain.
+     *
+     * @param string    $domain     Name of the domain.
+     *
+     * @return CredentialsListResponse
+     */
+    public function credentials_list($domain, $limit = 100, $skip = 0)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::integer($limit);
+        Assert::integer($skip);
+
+        $params = array(
+            'limit' => $limit,
+            'skip'  => $skip,
+        );
+
+        $response = $this->get(sprintf('/v3/domains/%s/credentials', $domain), $params);
+
+        return $this->serializer->deserialize($response, CredentialListResponse::class);
+    }
+
+    /**
+     * Create a new SMTP credential pair for the specified domain.
+     *
+     * @param string    $domain     Name of the domain.
+     * @param string    $login      SMTP Username.
+     * @param string    $pass       SMTP Password. Length min 5, max 32.
+     *
+     * @return CredentialResponse
+     */
+    public function new_credential($domain, $login, $pass)
+    {
+        Assert::stringNotEmpty($domain);
+        Assert::stringNotEmpty($login);
+        Assert::stringNotEmpty($pass);
+        Assert::lengthBetween($pass, 5, 32, 'SMTP password must be between 5 and 32 characters.');
```

Comment:
@DavidGarciaCat so I spent a good amount of time looking in to this... and as far as I can tell, there is not really a reason. I am going to be looking in to it some more shortly.